src/http-adapter/jquery/index.js

```diff
@@ -0,0 +1,80 @@
+/***
+ * JQueryHttpAdapter
+ * @module
+ */
+import HttpAdapter, { HTTP_METHOD } from '../index'
+import { ajax } from 'jquery'
+
+export default class JqueryHttpAdapter extends HttpAdapter {
+  /**
+   * The Http request method with jQuery implementation
+   * @private
+   * @param {Object} -  options passed via destructoring, which can contain url,
+   * method, headers, data, and isInternal properties
+   * @returns {Promise} Promise relating to request resolution.
+   */
+  async _makeRequest ({
+    url,
+    method,
+    headers,
+    data,
+    isInternal
+  }) {
+    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
+    // some instrumented Vue data)
+    if (data && [HTTP_METHOD.PATCH, HTTP_METHOD.POST].indexOf(method) > -1) {
+      data = JSON.stringify(data)
+    }
+
+    const requestInterceptor = this.requestInterceptor
+    const responseInterceptor = this.responseInterceptor
+
+    return new Promise((resolve, reject) => {
+      let request = {
+        url,
+        type: method,
+        headers,
+        beforeSend (xhr, options) {
+          if (isInternal) {
+            requestInterceptor(xhr, options)
+          }
+        },
```

Comment:
Does this mean that `ajaxSetup` will no longer have any effect on requests made through this adapter, or does jQuery execute both implementations of `beforeSend`?

Reply:
I need to verify this

Reply:
So it looks like the options are merged via [ajaxExtend](https://github.com/jquery/jquery/blob/2.2-stable/src/ajax.js#L115). If a 'local'`beforeSend` is defined, then that hook and only that hook is used. If no 'local' `beforeSend` exists, but one is defined globally via `ajaxSetup`, then that hook is used. In other words, practically speaking, `ajaxSetup` will no longer have any effect on requests made through this adapter in regards to `beforeSend`

Reply:
So your intent then is to roll this out as a new major version and have a corresponding groveco PR to reimplement the headers using `requestInterceptor`?